### ISSUE TITLE
Bundler could not find compatible versions for gem "claide"

### ISSUE THREAD

**Issue by ShpakovNikita** [fastlane/fastlane#20131]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I was using github actions for test flight deployment, but during gym phase there was a crash, with the following error:
```
[21:21:35]: Resolving Swift Package Manager dependencies...
[21:21:35]: $ xcodebuild -resolvePackageDependencies -scheme CluebySwift -project ./CluebySwift.xcodeproj -configuration Release
[21:21:36]: ▸ Command line invocation:
[21:21:36]: ▸     /Applications/Xcode_13.2.1.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -scheme CluebySwift -project ./CluebySwift.xcodeproj -configuration Release
[21:21:36]: ▸ User defaults from command line:
[21:21:36]: ▸     IDEPackageSupportUseBuiltinSCM = YES
[21:21:39]: ▸ Resolve Package Graph
[21:21:39]: ▸ Package.resolved file is corrupted or malformed; fix or delete the file to continue: unsupported schema version 2xcodebuild: error: Could not resolve package dependencies:
[21:21:39]: ▸   Package.resolved file is corrupted or malformed; fix or delete the file to continue: unsupported schema version 2

```
On the local device I have XCode 13.3 and manually calling `xcodebuild -resolvePackageDependencies -scheme CluebySwift -project ./CluebySwift.xcodeproj -configuration Release` works fine, but on the remote mac we can see Xcode_13.2.1, so I guess this is the main issue for this error. But if I'll add `xcversion(version: "~> 13.3.0")` to the Fastfile, it will require adding `gem "xcode-install"`, which will result in the final issue:

```
Resolving dependencies...
Bundler could not find compatible versions for gem "claide":
  In snapshot (Gemfile.lock):
    claide (= 1.1.0)
  In Gemfile:
    claide (= 1.1.0)
    xcode-install was resolved to 0.9.2, which depends on
      claide (~> 0.8.1)
Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
bundler: failed to load command: fastlane (/usr/local/lib/ruby/gems/2.7.0/bin/fastlane)
/usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.9/lib/bundler/resolver.rb:57:in `rescue in start': Bundler could not find compatible versions for gem "claide": (Bundler::VersionConflict)
  In snapshot (Gemfile.lock):
    claide (= 1.1.0)
  In Gemfile:
    claide (= 1.1.0)
    xcode-install was resolved to 2.8.0, which depends on
      claide (>= 0.9.1, < 1.1.0)
Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

Also, this is the output of the `bundle update`:

```
Resolving dependencies...
Bundler could not find compatible versions for gem "claide":
  In snapshot (Gemfile.lock):
    claide (>= 1.1.0)

  In Gemfile:
    xcode-install was resolved to 0.9.2, which depends on
      claide (~> 0.8.1)

    fastlane was resolved to 2.205.1, which depends on
      xcodeproj (>= 1.13.0, < 2.0.0) was resolved to 1.21.0, which depends on
        claide (>= 1.0.2, < 2.0)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

Bundler could not find compatible versions for gem "faraday":
  In snapshot (Gemfile.lock):
    faraday (>= 1.10.0)

  In Gemfile:
    fastlane was resolved to 2.205.1, which depends on
      faraday-cookie_jar (~> 0.0.6) was resolved to 0.0.7, which depends on
        faraday (>= 0.8.0)

    fastlane was resolved to 2.205.1, which depends on
      faraday (~> 1.0)

    fastlane was resolved to 2.205.1, which depends on
      google-cloud-storage (~> 1.31) was resolved to 1.36.1, which depends on
        google-cloud-core (~> 1.6) was resolved to 1.6.0, which depends on
          google-cloud-env (~> 1.0) was resolved to 1.6.0, which depends on
            faraday (>= 0.17.3, < 3.0)

    fastlane was resolved to 2.205.1, which depends on
      google-cloud-storage (~> 1.31) was resolved to 1.36.1, which depends on
        googleauth (>= 0.16.2, < 2.a) was resolved to 1.1.2, which depends on
          faraday (>= 0.17.3, < 3.a)

    fastlane was resolved to 2.205.1, which depends on
      google-cloud-storage (~> 1.31) was resolved to 1.36.1, which depends on
        googleauth (>= 0.16.2, < 2.a) was resolved to 1.1.2, which depends on
          signet (>= 0.16, < 2.a) was resolved to 0.16.1, which depends on
            faraday (>= 0.17.5, < 3.0)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

And I'm stuck with this issue right now. Any help would be appreciated! Maybe I'm doing something wrong in another place. Thanks!

##### Command executed

Fastline github action maierj/fastlane-action@v1.4.0
A specific error was caused by `bundle install` I suppose 

### Environment

<details>
  <pre>
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                      |
| --------------------------- | ---------------------------------------------------------- |
| OS                          | 12.0.1                                                     |
| Ruby                        | 3.1.1                                                      |
| Bundler?                    | false                                                      |
| Git                         | git version 2.32.0 (Apple Git-132)                         |
| Installation Source         | /opt/homebrew/Cellar/fastlane/2.205.0/libexec/bin/fastlane |
| Host                        | macOS 12.0.1 (21A559)                                      |
| Ruby Lib Dir                | /opt/homebrew/Cellar/ruby/3.1.1/lib                        |
| OpenSSL Version             | OpenSSL 1.1.1m  14 Dec 2021                                |
| Is contained                | false                                                      |
| Is homebrew                 | true                                                       |
| Is installed via Fabric.app | false                                                      |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                |
| Xcode Version               | 13.3                                                       |
| Swift Version               | 5.6                                                        |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)
xcversion(version: "~> 13.3.0")

DEVELOPER_APP_ID = ENV["DEVELOPER_APP_ID"]
DEVELOPER_APP_IDENTIFIER = ENV["DEVELOPER_APP_IDENTIFIER"]
PROVISIONING_PROFILE_SPECIFIER = ENV["PROVISIONING_PROFILE_SPECIFIER"]
TEMP_KEYCHAIN_USER = ENV["TEMP_KEYCHAIN_USER"]
TEMP_KEYCHAIN_PASSWORD = ENV["TEMP_KEYCHAIN_PASSWORD"]
APPLE_ISSUER_ID = ENV["APPLE_ISSUER_ID"]
APPLE_KEY_ID = ENV["APPLE_KEY_ID"]
APPLE_KEY_CONTENT = ENV["APPLE_KEY_CONTENT"]
GIT_AUTHORIZATION = ENV["GIT_AUTHORIZATION"]

def delete_temp_keychain(name)
  delete_keychain(
    name: name
  ) if File.exist? File.expand_path("~/Library/Keychains/#{name}-db")
end

def create_temp_keychain(name, password)
  create_keychain(
    name: name,
    password: password,
    unlock: false,
    timeout: 0
  )
end

def ensure_temp_keychain(name, password)
  delete_temp_keychain(name)
  create_temp_keychain(name, password)
end

platform :ios do
  lane :closed_beta do
    keychain_name = TEMP_KEYCHAIN_USER
    keychain_password = TEMP_KEYCHAIN_PASSWORD
    ensure_temp_keychain(keychain_name, keychain_password)

    api_key = app_store_connect_api_key(
      key_id: APPLE_KEY_ID,
      issuer_id: APPLE_ISSUER_ID,
      key_content: APPLE_KEY_CONTENT,            
      duration: 1200,            
      in_house: false
    )
    
    increment_build_number(xcodeproj: "CluebySwift.xcodeproj")

    match(
      type: 'appstore',
      app_identifier: "#{DEVELOPER_APP_IDENTIFIER}",
      git_basic_authorization: Base64.strict_encode64(GIT_AUTHORIZATION),
      readonly: true,
      keychain_name: keychain_name,
      keychain_password: keychain_password,
      api_key: api_key
    )

    gym(
      configuration: "Release",
      scheme: "CluebySwift",
      export_method: "app-store",
      export_options: {
        provisioningProfiles: { 
            DEVELOPER_APP_ID => PROVISIONING_PROFILE_SPECIFIER
        }
      }
    )

    pilot(
      apple_id: "#{DEVELOPER_APP_ID}",
      app_identifier: "#{DEVELOPER_APP_IDENTIFIER}",
      skip_waiting_for_build_processing: true,
      skip_submission: true,
      distribute_external: false,
      notify_external_testers: false,
      ipa: "./CluebySwift.ipa"
    )

    delete_temp_keychain(keychain_name)
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("app.ios.swiftui.CluebySwift") # The bundle identifier of your app
apple_id("shpakovnikita1998@gmail.com") # Your Apple email address
team_id("75589KQ95H")


# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.205.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| error_highlight                 | 0.3.0        |
| did_you_mean                    | 1.6.1        |
| atomos                          | 0.1.3        |
| rexml                           | 3.2.5        |
| CFPropertyList                  | 3.0.5        |
| claide                          | 1.1.0        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.21.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.6.0        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| optparse                        | 0.1.1        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-multipart               | 1.0.3        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.0       |
| unf_ext                         | 0.0.8.1      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.6        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.6.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.1        |
| dotenv                          | 2.7.6        |
| bundler                         | 2.3.7        |
| simctl                          | 1.6.8        |
| jwt                             | 2.3.0        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.2        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.2        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.16.1       |
| googleauth                      | 1.1.2        |
| httpclient                      | 2.8.3        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.4.2        |
| google-apis-playcustomapp_v1    | 0.7.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.10.0       |
| google-apis-storage_v1          | 0.11.0       |
| rake                            | 13.0.6       |
| digest-crc                      | 0.6.4        |
| google-cloud-storage            | 1.36.1       |
| emoji_regex                     | 3.2.3        |
| jmespath                        | 1.6.1        |
| aws-eventstream                 | 1.2.0        |
| aws-sigv4                       | 1.4.0        |
| aws-sdk-core                    | 3.130.0      |
| aws-sdk-kms                     | 1.55.0       |
| aws-sdk-s3                      | 1.113.0      |
| set                             | 1.0.2        |
| excon                           | 0.92.1       |
| google-apis-androidpublisher_v3 | 0.17.0       |
| google-cloud-env                | 1.6.0        |
| aws-partitions                  | 1.570.0      |
| forwardable                     | 1.3.2        |
| logger                          | 1.5.0        |
| pathname                        | 0.2.0        |
| shellwords                      | 0.1.0        |
| cgi                             | 0.3.1        |
| date                            | 3.2.2        |
| timeout                         | 0.2.0        |
| stringio                        | 3.0.1        |
| securerandom                    | 0.1.1        |
| uri                             | 0.11.0       |
| openssl                         | 3.0.0        |
| digest                          | 3.1.0        |
| io-nonblock                     | 0.1.0        |
| ipaddr                          | 1.2.4        |
| io-wait                         | 0.2.1        |
| zlib                            | 2.1.1        |
| resolv                          | 0.2.1        |
| time                            | 0.2.0        |
| open-uri                        | 0.2.0        |
| mutex_m                         | 0.1.1        |
| net-protocol                    | 0.1.2        |
| net-http                        | 0.2.0        |
| ostruct                         | 0.5.2        |
| english                         | 0.7.1        |
| erb                             | 2.2.3        |
| strscan                         | 3.0.1        |
| abbrev                          | 0.1.0        |
| io-console                      | 0.5.11       |
| tempfile                        | 0.1.2        |
| delegate                        | 0.2.0        |
| fileutils                       | 1.6.0        |
| tmpdir                          | 0.1.2        |
| base64                          | 0.1.1        |
| singleton                       | 0.1.1        |
| open3                           | 0.1.1        |
| nkf                             | 0.1.1        |
| prettyprint                     | 0.1.1        |
| pp                              | 0.3.0        |
| find                            | 0.1.1        |
| yaml                            | 0.2.0        |
| psych                           | 4.0.3        |
</details>

*generated on:* **2022-04-05**
</details>
 </pre>
</details>

**Comment by kambala-decapitator** [comment]
also stumbled on this. What helped me was deleting `Gemfile.lock` and running `bundle update`. After that `xcode-install` gem was correctly resolved to version 2.8.0.

**Comment by Farr69** [comment]
Please update the issue to include the output of fastlane env.

**Comment by ShpakovNikita** [comment]
Thank you, @kambala-decapitator. Actually it seems like it did something in my repo - but right now I have the following issue:
<pre>
Bundle complete! 2 Gemfile dependencies, 91 gems now installed.
[127](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:127)
Use `bundle info [gemname]` to see where a bundled gem is installed.
[128](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:128)
[23:31:34]: Sending anonymous analytics information
[129](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:129)
[23:31:34]: Learn more at https://docs.fastlane.tools/#metrics
[130](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:130)
[23:31:34]: No personal or sensitive data is sent.
[131](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:131)
[23:31:34]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[132](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:132)
[23:31:34]: ------------------------------
[133](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:133)
[23:31:34]: --- Step: default_platform ---
[134](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:134)
[23:31:34]: ------------------------------
[135](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:135)
[23:31:34]: -----------------------
[136](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:136)
[23:31:34]: --- Step: xcversion ---
[137](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:137)
[23:31:34]: -----------------------
[138](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:138)

[139](https://github.com/clueby/ios.clueby.com/runs/5825498319?check_suite_focus=true#step:3:139)
[!] Cannot find an installed Xcode satisfying '~> 13.3.0'
</pre>

@Farr69 there is output under Environment tab if I did everything right (updated after @kambala-decapitator  advise)

**Comment by kambala-decapitator** [comment]
@ShpakovNikita the error seems quite obvious: you don't have Xcode 13.3 installed on your machine.

@Farr69 I think the steps to reproduce the issue are the following (at least that was my case):

1. add fastlane to Gemfile
2. run `bundle install`
3. add xcode-install to Gemfile
4. run `bundle install` => error

**Comment by Alokcbre** [comment]
I was also getting the same issue and just added =< 1.1.0 instead of < 1.1.0

xcode-install (2.8.0)
      claide (>= 0.9.1, =< 1.1.0) 
      
      Just changed <1.1.0 to =< 1.1.0

**Comment by AlokPandey-IOS** [comment]
I was also getting the same issue and just added =< 1.1.0 instead of < 1.1.0

xcode-install (2.8.0)
claide (>= 0.9.1, =< 1.1.0)

  Just changed <1.1.0 to =< 1.1.0

**Comment by kambala-decapitator** [comment]
> @Farr69 I think the steps to reproduce the issue are the following (at least that was my case):
> 
> 1. add fastlane to Gemfile
> 2. run `bundle install`
> 3. add xcode-install to Gemfile
> 4. run `bundle install` => error

just had this error again. Adding both fastlane and xcode-install at once produces the same issue.

Tried to add xcode-install with explicit version: `gem "xcode-install", "~> 2.8"`, and now the error is clear:

```
Bundler could not find compatible versions for gem "claide":
  In snapshot (Gemfile.lock):
    claide (= 1.1.0)

  In Gemfile:
    cocoapods was resolved to 1.11.3, which depends on
      claide (>= 1.0.2, < 2.0)

    xcode-install (~> 2.8) was resolved to 2.8.0, which depends on
      claide (>= 0.9.1, < 1.1.0)

Deleting your Gemfile.lock file and running `bundle install` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

However, it's still unclear why adding xcode-install without explicit version picks 0.9.2 instead of the latest.

**Comment by kambala-decapitator** [comment]
FYI this issue should have been solved with [xcode-install v2.8.1](https://github.com/xcpretty/xcode-install/releases/tag/v2.8.1):

> Remove CLAide version restriction by @triplef in https://github.com/xcpretty/xcode-install/pull/454
